NewForce-Cohort-9/tabloidmvc-nacua-matata
# Df branch1 - #23 View All User Profiles

#23 View All User Profiles - https://trello.com/c/yIDLpjA5

**Description**
This PR implements a new feature that allows Admin users to view a list of all User Profiles in the system. The list is displayed on the "User Profiles" page, which is accessible only to users with a UserTypeId of 1 (Admin). The "User Profiles" link is visible in the navigation menu only to logged-in admin users. The list of user profiles is ordered alphabetically by DisplayName, providing a clear and organized view of all users in the system.

**Fixes**: None (No specific issue number associated with this PR)

**Type of Change**
New feature (non-breaking change which adds functionality)

**Motivation and Context**
This change enhances the administrative capabilities of the application by ensuring that only admin users can access and view the list of all user profiles. This provides admins with the necessary tools to manage user accounts effectively while maintaining system security by restricting access to this feature.

**Dependencies**
No new dependencies are required for this change.

**Testing Instructions**

1. Set Up Your Environment:

- Ensure your current work is committed: git add . and git commit -m "Your message"
- Switch to the main branch: git checkout main
- Fetch the latest changes: git fetch --all
- Checkout the my branch: git checkout df-branch1
- If necessary, pull the latest changes on the branch: git pull origin df-branch1

2. Running the Application:

- Launch the application by clicking the "Execute with Debugger" button.
- Log in as an Admin: Use the credentials [admin@example.com](mailto:admin@example.com).
- Verify Menu: Ensure that the "User Profiles" link is visible in the navigation menu.
- Accessing the User Profiles: Click on the "User Profiles" link and verify that you are directed to the User Profiles list page.
- Verify List: Ensure the list displays all user profiles, ordered alphabetically by DisplayName.

**Checklist**
